src/test/unit/EigenPodUnit.t.sol

```diff
@@ -275,4 +275,143 @@ contract EigenPodUnitTests is EigenPodTests {
+        cheats.assume(nonPodOwner != podOwner);
+        cheats.startPrank(podOwner);
+        IEigenPod newPod = eigenPodManager.getPod(podOwner);
+        cheats.expectEmit(true, true, true, true, address(newPod));
+        emit EigenPodStaked(pubkey);
+        eigenPodManager.stake{value: stakeAmount}(pubkey, signature, depositDataRoot);
+        cheats.stopPrank();
+        
+        uint256 amount = 32 ether;
+
+        cheats.store(address(newPod), bytes32(uint256(52)), bytes32(0));
+
+        cheats.startPrank(nonPodOwner);
+        cheats.expectRevert(bytes("EigenPod.onlyEigenPodOwner: not podOwner"));
+        newPod.withdrawBeforeRestaking();
+        cheats.stopPrank();  
+    }
+
+    function testDelayedWithdrawalIsCreatedByWithdrawBeforeRestaking(uint256 amount) external {
+        cheats.startPrank(podOwner);
+        IEigenPod newPod = eigenPodManager.getPod(podOwner);
+        cheats.expectEmit(true, true, true, true, address(newPod));
+        emit EigenPodStaked(pubkey);
+        eigenPodManager.stake{value: stakeAmount}(pubkey, signature, depositDataRoot);
+        cheats.stopPrank();
+        
+        uint256 amount = 32 ether;
+
+        cheats.store(address(newPod), bytes32(uint256(52)), bytes32(0));
+        cheats.deal(address(this), amount);
+        // simulate a withdrawal processed on the beacon chain, pod balance goes to 32 ETH
+        Address.sendValue(payable(address(newPod)), amount);
+        require(newPod.nonBeaconChainETHBalanceWei() == amount, "nonBeaconChainETHBalanceWei should be 32 ETH");
+
+        cheats.startPrank(podOwner);
+        newPod.withdrawBeforeRestaking();
+        cheats.stopPrank();
+
+        require(_getLatestDelayedWithdrawalAmount(podOwner) == amount, "Payment amount should be stake amount");
+    }
+
+    function testFullWithdrawalsProvenOutOfOrder(bytes32 pubkeyHash, uint64 withdrawalAmount) external {
+        uint64 timestamp = 5;
+        uint64 lesserTimestamp = 3;
+        _deployInternalFunctionTester();
+        IEigenPod.ValidatorInfo memory validatorInfo = IEigenPod.ValidatorInfo({
+            validatorIndex: 0,
+            restakedBalanceGwei: 0,
+            mostRecentBalanceUpdateTimestamp: 0,
+            status: IEigenPod.VALIDATOR_STATUS.ACTIVE
+        });
+        podInternalFunctionTester.processFullWithdrawal(0, pubkeyHash, timestamp += 5 , podOwner, withdrawalAmount, validatorInfo);
+        IEigenPod.ValidatorInfo memory info = podInternalFunctionTester.validatorPubkeyHashToInfo(pubkeyHash);
+
+        podInternalFunctionTester.processFullWithdrawal(0, pubkeyHash, lesserTimestamp , podOwner, withdrawalAmount, podInternalFunctionTester.validatorPubkeyHashToInfo(pubkeyHash));
+
+        IEigenPod.ValidatorInfo memory info2 = podInternalFunctionTester.validatorPubkeyHashToInfo(pubkeyHash);
+
+        require(info2.mostRecentBalanceUpdateTimestamp == timestamp, "mostRecentBalanceUpdateTimestamp should not have been updated");
+    }
```

Comment:
this should have a comment or something at least explaining what the intended behavior is here.
The naming makes it sound like maybe it should revert to me (which it shouldn't)

Reply:
this test is no longer relevant